core/trino-main/src/main/java/io/trino/operator/window/pattern/ArgumentComputation.java

```diff
@@ -0,0 +1,120 @@
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.operator.window.pattern;
+
+import io.trino.operator.DriverYieldSignal;
+import io.trino.operator.Work;
+import io.trino.operator.project.PageProjection;
+import io.trino.spi.Page;
+import io.trino.spi.block.Block;
+import io.trino.spi.connector.ConnectorSession;
+import io.trino.spi.function.WindowIndex;
+import io.trino.spi.type.Type;
+
+import java.util.List;
+import java.util.function.Predicate;
+import java.util.function.Supplier;
+
+import static io.airlift.slice.Slices.utf8Slice;
+import static io.trino.operator.project.SelectedPositions.positionsRange;
+import static io.trino.operator.window.pattern.PhysicalValuePointer.CLASSIFIER;
+import static io.trino.operator.window.pattern.PhysicalValuePointer.MATCH_NUMBER;
+import static io.trino.operator.window.pattern.UnimplementedBlock.UNIMPLEMENTED_BLOCK;
+import static io.trino.spi.predicate.Utils.nativeValueToBlock;
+import static io.trino.spi.type.BigintType.BIGINT;
+import static io.trino.spi.type.VarcharType.VARCHAR;
+import static java.util.Objects.requireNonNull;
+
+public class ArgumentComputation
+{
+    private final PageProjection projection;
+    private final Type outputType;
+    private final List<Integer> inputChannels;
+    private final Predicate<Integer> appliesToLabel;
+    private final List<String> labelNames;
+    private final ConnectorSession session;
+
+    private ArgumentComputation(PageProjection projection, Type outputType, List<Integer> inputChannels, Predicate<Integer> appliesToLabel, List<String> labelNames, ConnectorSession session)
+    {
+        this.projection = requireNonNull(projection, "projection is null");
+        this.outputType = requireNonNull(outputType, "outputType is null");
+        this.inputChannels = requireNonNull(inputChannels, "inputChannels is null");
+        this.appliesToLabel = requireNonNull(appliesToLabel, "appliesToLabel is null");
+        this.labelNames = requireNonNull(labelNames, "labelNames is null");
+        this.session = requireNonNull(session, "session is null");
+    }
+
+    public Type getOutputType()
+    {
+        return outputType;
+    }
+
+    public Block compute(int position, WindowIndex windowIndex, int label, long matchNumber)
+    {
+        // This aggregation argument is needed only for rows with certain labels. If the label assigned to the row is not interesting from the aggregation's viewpoint, we can skip computations.
+        if (!appliesToLabel.test(label)) {
+            return UNIMPLEMENTED_BLOCK;
```

Comment:
When would the code hit this path? Why not just throw an exception here?

Reply:
Currently this is simplified, and the argument computation is only called for applicable rows.